### PR TITLE
Fix cosine similarity when missing simd alignment

### PR DESCRIPTION
**Pull request by changhiskhan**
also a bunch of formatting fixes

### CODE REVIEW

File: rust/src/linalg/cosine.rs [hunk 1]
```diff
@@ -111,14 +113,31 @@ mod x86_64 {
...
+                let mut y_sq_sum = add_f32_register(y_sq);
+                y_sq_sum += sq_scalar(&y_vector[len..]);
+                1.0 - dotprod / (x_norm * y_sq_sum.sqrt())
             }
         }
     }
 }
 
+#[inline]
+fn dotprod_scalar<T: Real + Sum>(x: &[T], y: &[T]) -> T {
```
Comment (eddyxu) [inline rust/src/linalg/cosine.rs:128]: Lets call https://github.com/eto-ai/lance/blob/main/rust/src/linalg/dot.rs ?

Will add SIMD for dot product later.

File: rust/src/linalg/cosine.rs [hunk 2]
```diff
@@ -92,6 +92,8 @@ pub fn cosine_distance_batch(from: &[f32], to: &[f32], dimension: usize) -> Arc<
 
 #[cfg(target_arch = "x86_64")]
 mod x86_64 {
+    use super::dotprod_scalar;
+    use super::sq_scalar;
     use std::arch::x86_64::*;
```
Comment (eddyxu) [inline rust/src/linalg/cosine.rs:97]: sort import by sections? `std / third party / lance` with a blank line in between.

File: rust/src/linalg/cosine.rs [hunk 3]
```diff
@@ -111,14 +113,31 @@ mod x86_64 {
...
     }
 }
 
+#[inline]
+fn dotprod_scalar<T: Real + Sum>(x: &[T], y: &[T]) -> T {
+    x.iter().zip(y.iter()).map(|(a, b)| (a.mul(*b))).sum::<T>()
+}
+
+#[inline]
+fn sq_scalar<T: Real + Sum>(y: &[T]) -> T {
```
Comment (eddyxu) [inline rust/src/linalg/cosine.rs:133]: Extract this to a new routine? or just use normalize().pow(2)?

File: rust/src/linalg/cosine.rs (hunk 1 above)
Comment (changhiskhan) [reply rust/src/linalg/cosine.rs:128]: fixed

File: rust/src/linalg/cosine.rs (hunk 2 above)
Comment (changhiskhan) [reply rust/src/linalg/cosine.rs:97]: fixed

File: rust/src/linalg/cosine.rs (hunk 3 above)
Comment (changhiskhan) [reply rust/src/linalg/cosine.rs:133]: fixed

**Review by eddyxu — APPROVED** [review]
LGTM. Thanks

**Pull request merged**